File: River/River.Body/Ahoy/Messages/Message.py


class Message():
    def __init__(self, channel, command, dataType, data):
        self.channel = channel
        self.command = command
        self.dataType = dataType
        self.data = data

    def __str__(self):
        return "channel: {}, command: {}, data type: {}, data length: {} data: {}".format(self.channel, self.command, self.dataType, len(self.data), self.data)

File: River/River.Droid/Assets/Packages/Ahoy.Net/Scripts/Messages/Message.cs



namespace Ahoy.Net
{

    public class Message
    {
        public byte channel;
        public byte command;
        public DataType dataType;

        public Message(byte _channel, byte _command, DataType _dataType)
        {
            channel = _channel;
            command = _command;
            dataType = _dataType;
        }

        public override string ToString()
        {
            return $"channel: {channel}, command: {command}, data type: {dataType}";
        }
    }

    public class Message<T> : Message
    {
        public T[] data;
        public Message(byte _channel, byte _command, DataType _dataType, T[] _data)
        : base(_channel, _command, _dataType)
        {
            data = _data;
        }


        public override string ToString()
        {
            return base.ToString() + $", data length: {data.Length}, data: {data.ElementsToString()}";
        }
    }


}